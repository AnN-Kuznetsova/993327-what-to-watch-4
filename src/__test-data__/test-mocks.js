const promoMovie = {
  title: `The Grand Budapest Hotel`,
  smallPictureUrl: ``,
  backgroundUrl: `img/bg-the-grand-budapest-hotel.jpg`,
  posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
  previewUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  genres: [`Drama`, `Comedy`],
  releaseDate: new Date(2014, 0),
  description: [
    `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy,
    becomes Gustave's friend and protege.`,
    `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay
    there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  ],
  directors: [`Wes Andreson`],
  starring: [`Bill Murray`,
    `Edward Norton`,
    `Jude Law`,
    `Willem Dafoe`,
    `Saoirse Ronan`,
    `Tony Revoloru`,
    `Tilda Swinton`,
    `Tom Wilkinson`,
    `Owen Wilkinson`,
    `Adrien Brody`,
    `Ralph Fiennes`,
    `Jeff Goldblum`,
  ],
  runTime: 99,
  reviews: [],
  rating: {
    score: 8.9,
    totalVotes: 240,
  },
};

const films = [
  {
    title: `Bohemian Rhapsody`,
    smallPictureUrl: `img/bohemian-rhapsody.jpg`,
    backgroundUrl: `img/bohemian-rhapsody.jpg`,
    posterUrl: `img/bohemian-rhapsody.jpg`,
    previewUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genres: [`Drama`, `Biography`],
    releaseDate: new Date(2018, 0),
    description: [
      `Сюжет картины охватывает период 1970—1985 года, рассказывая о жизни, творчестве и отношениях внутри группы Queen, а также о личной жизни фронтмена
      группы — Фредди Меркьюри.`,
      `В 1970 году Фаррух Булсара, студент-дизайнер и грузчик багажа в аэропорту, посещает концерт группы Smile. После шоу Фаррух встречается с членами
      группы и узнаёт, что группу только что покинул вокалист. Недолго думая, он предлагает себя в качестве нового вокалиста, и его предложение было одобрено
      участниками группы — Брайаном Мэем и Роджером Тейлором. Фаррух берёт себе сценический псевдоним Фредди Меркьюри. Новый коллектив с подачи Фредди получает
      название Queen и начинает гастрольную деятельность по Великобритании. `
    ],
    directors: [
      `Брайан Сингер`,
      `Декстер Флетчер`,
    ],
    starring: [
      `Рами Малек`,
      `Люси Бойнтон`,
      `Гвилим Ли`,
      `Бен Харди`,
      `Джозеф Маццелло`,
      `Эйдан Гиллен`,
      `Том Холландер`,
      `Аллен Лич`,
      `Майк Майерс`,
    ],
    runTime: 134,
    reviews: [],
    rating: {
      score: 8.0,
      totalVotes: 350,
    },
  },

  {
    title: `Aviator`,
    smallPictureUrl: `img/aviator.jpg`,
    backgroundUrl: `img/aviator.jpg`,
    posterUrl: `img/aviator.jpg`,
    previewUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genres: [`Drama`, `Biography`],
    releaseDate: new Date(2004, 0),
    description: [
      `Получив от отца небольшую фабрику, Говард Хьюз превратил ее в гигантское, фантастически прибыльное предприятие. Став владельцем огромной кинокомпании,
      он снял самый дорогой для своего времени фильм и покорил сердца прелестнейших голливудских актрис.`,
      `Ему принадлежали самые престижные казино Лас-Вегаса и он установил рекорд скоростных полетов, приобрел вторую по величине коммерческую авиакомпанию…`
    ],
    directors: [`Мартин Скорсезе`],
    starring: [
      `Леонардо ДиКаприо`,
      `Кейт Бланшетт`,
      `Мэтт Росс`,
      `Джон Си Райли`,
      `Алан Алда`,
      `Кейт Бекинсейл`,
      `Алек Болдуин`,
      `Иэн Холм`,
      `Адам Скотт`,
      `Дэнни Хьюстон`,
    ],
    runTime: 163,
    reviews: [],
    rating: {
      score: 7.6,
      totalVotes: 120,
    },
  },

  {
    title: `Revenant`,
    smallPictureUrl: `img/revenant.jpg`,
    backgroundUrl: `img/revenant.jpg`,
    posterUrl: `img/revenant.jpg`,
    previewUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genres: [`Drama`, `Biography`, `Adventure`, `Western`, `Action`],
    releaseDate: new Date(2015, 0),
    description: [
      `Охотник Хью Гласс серьезно ранен на неизведанных просторах американского Дикого Запада. Товарищ Хью по отряду покорителей новых земель Джон Фицжеральд
      предательски оставляет его умирать в одиночестве. Теперь у Гласса осталось только одно оружие — его сила воли.`,
      `Он готов бросить вызов первобытной природе, суровой зиме и враждебным племенам индейцев, только чтобы выжить и отомстить Фицжеральду.`,
    ],
    directors: [`Алехандро Гонсалес Иньярриту`],
    starring: [
      `Леонардо ДиКаприо`,
      `Том Харди`,
      `Донал Глисон`,
      `Уилл Поултер`,
      `Форрест Гудлак`,
      `Пол Андерсон`,
      `Кристоффер Йонер`,
      `Джошуа Бёрдж`,
      `Дуан Ховард`,
      `Мила Нахеко`,
    ],
    runTime: 156,
    reviews: [],
    rating: {
      score: 7.8,
      totalVotes: 263,
    },
  },

  {
    title: `Johnny English`,
    smallPictureUrl: `img/johnny-english.jpg`,
    backgroundUrl: `img/johnny-english.jpg`,
    posterUrl: `img/johnny-english.jpg`,
    previewUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genres: [`Comedy`, `Action`, `Comedy`, `Adventure`],
    releaseDate: new Date(2003, 0),
    description: [
      `История самого некомпетентного и недалекого Британского дипломата, которого только можно представить, которого ошибочно принимают за самого знаменитого
      и опасного шпиона Великобритании, Джонни Инглиша. Из-за этой путаницы герой оказывается вовлеченным в смертельную схватку с авторами очередного
      дьявольского плана.`,
    ],
    directors: [`Питер Хауит`],
    starring: [
      `Роуэн Эткинсон`,
      `Натали Имбрулья`,
      `Бен Миллер`,
      `Джон Малкович`,
      `Тим Пиготт-Смит`,
      `Кевин МакНэлли`,
      `Оливер Форд Дейвис`,
      `Дуглас МакФерран`,
      `Таша ди Вашконселуш`,
      `Грег Уайз      `,
    ],
    runTime: 84,
    reviews: [],
    rating: {
      score: 6.6,
      totalVotes: 361,
    },
  },

  {
    title: `Shutter Island`,
    smallPictureUrl: `img/shutter-island.jpg`,
    backgroundUrl: `img/shutter-island.jpg`,
    posterUrl: `img/shutter-island.jpg`,
    previewUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genres: [`Drama`, `Thriller`, `Detective`],
    releaseDate: new Date(2009, 0),
    description: [
      `Два американских судебных пристава отправляются на один из островов в штате Массачусетс, чтобы расследовать исчезновение пациентки клиники для умалишенных
      преступников. При проведении расследования им придется столкнуться с паутиной лжи, обрушившимся ураганом и смертельным бунтом обитателей клиники.`,
    ],
    directors: [`Мартин Скорсезе`],
    starring: [
      `Леонардо ДиКаприо`,
      `Марк Руффало`,
      `Бен Кингсли`,
      `Макс фон Сюдов`,
      `Мишель Уильямс`,
      `Эмили Мортимер`,
      `Патриша Кларксон`,
      `Джеки Эрл Хейли`,
      `Тед Левайн`,
      `Джон Кэрролл Линч      `,
    ],
    runTime: 138,
    reviews: [],
    rating: {
      score: 8.5,
      totalVotes: 499,
    },
  },

  {
    title: `Pulp Fiction`,
    smallPictureUrl: `img/pulp-fiction.jpg`,
    backgroundUrl: `img/pulp-fiction.jpg`,
    posterUrl: `img/pulp-fiction.jpg`,
    previewUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genres: [`Thriller`, `Comedy`, `Crime`],
    releaseDate: new Date(1994, 0),
    description: [
      `Двое бандитов Винсент Вега и Джулс Винфилд ведут философские беседы в перерывах между разборками и решением проблем с должниками криминального босса
      Марселласа Уоллеса.`,
      `В первой истории Винсент проводит незабываемый вечер с женой Марселласа Мией. Во второй рассказывается о боксёре Бутче Кулидже, купленном Уоллесом,
      чтобы сдать бой. В третьей истории Винсент и Джулс по нелепой случайности попадают в неприятности.`,
    ],
    directors: [`Квентин Тарантино`],
    starring: [
      `Джон Траволта`,
      `Сэмюэл Л. Джексон`,
      `Брюс Уиллис`,
      `Ума Турман`,
      `Винг Реймз`,
      `Тим Рот`,
      `Харви Кейтель`,
      `Квентин Тарантино`,
      `Питер Грин`,
      `Аманда Пламмер      `,
    ],
    runTime: 154,
    reviews: [],
    rating: {
      score: 8.6,
      totalVotes: 454,
    },
  },

  {
    title: `Seven Years in Tibet`,
    smallPictureUrl: `img/seven-years-in-tibet.jpg`,
    backgroundUrl: `img/seven-years-in-tibet.jpg`,
    posterUrl: `img/seven-years-in-tibet.jpg`,
    previewUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genres: [`Drama`, `Adventure`, `Military`, `Biography`, `History`],
    releaseDate: new Date(1997, 0),
    description: [
      `Генрих Харрер, офицер Рейха, человек, у которого было все — успех, красота, известность. Но ему было мало земной славы. Оставив дома беременную жену,
      он отправляется на покорение самого неприступного пика в Гималаях. Этот подвиг должен увековечить его имя.`,
      `Однако впереди его ждет не успешное восхождение, а смертельная опасность, вражеский плен, долгие скитания… Наконец он добирается до таинственного
      тибетского города Лхаса, где ему предстоит провести долгих семь лет, и где молодой Далай Лама делает его своим приближенным. И здесь, в стране, лежащей за
      границами цивилизации, ему придется пережить события, которые навсегда изменят его жизнь…`,
    ],
    directors: [`Жан-Жак Анно`],
    starring: [
      `Брэд Питт`,
      `Дэвид Тьюлис`,
      `Б.Д. Вонг`,
      `Мако`,
      `Дэнни Дензонгпа`,
      `Виктор Вонг`,
      `Ингеборга Дапкунайте`,
      `Цзямъян Цзямцо Ванчук`,
      `Лакпа Тсамшо`,
      `Джетсун Пема      `,
    ],
    runTime: 136,
    reviews: [],
    rating: {
      score: 7.8,
      totalVotes: 390,
    },
  },

  {
    title: `Snatch`,
    smallPictureUrl: `img/snatch.jpg`,
    backgroundUrl: `img/snatch.jpg`,
    posterUrl: `img/snatch.jpg`,
    previewUrl: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    genres: [`Action`, `Comedy`, `Crime`],
    releaseDate: new Date(2000, 0),
    description: [
      `Четырехпалый Френки должен был переправить краденый алмаз из Англии в США своему боссу Эви. Но вместо этого герой попадает в эпицентр больших неприятностей.
      Сделав ставку на подпольном боксерском поединке, Френки попадает в круговорот весьма нежелательных событий.`,
      `Вокруг героя и его груза разворачивается сложная интрига с участием множества колоритных персонажей лондонского дна — русского гангстера, троих незадачливых
      грабителей, хитрого боксера и угрюмого громилы грозного мафиози. Каждый норовит в одиночку сорвать Большой Куш.`,
    ],
    directors: [`Гай Ричи`],
    starring: [
      `Джейсон Стэйтем`,
      `Стивен Грэм`,
      `Брэд Питт`,
      `Алан Форд`,
      `Робби Ги`,
      `Ленни Джеймс`,
      `Эд`,
      `Деннис Фарина`,
      `Раде Шербеджия`,
      `Винни Джонс      `,
    ],
    runTime: 104,
    reviews: [],
    rating: {
      score: 8.5,
      totalVotes: 343,
    },
  },
];

const VideoPlayerMode = {
  PREVIEW: `preview`,
  FULL_SCREEN: `full-screen`,
};

const VideoPlayerStatus = {
  ON_AUTOPLAY: `on-autoplay`,
  ON_PLAY: `on-play`,
  ON_PAUSE: `on-pause`,
  ON_RESET: `on-reset`,
};

const genreNames = [
  `All genres`,
  `Comedies`,
  `Crime`,
  `Documentary`,
  `Dramas`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thrillers`,
];


export {
  VideoPlayerMode,
  VideoPlayerStatus,
  genreNames,
  promoMovie,
  films,
};
