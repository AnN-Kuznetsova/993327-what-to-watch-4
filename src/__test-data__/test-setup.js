import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";

Enzyme.configure({
  adapter: new Adapter(),
});

const noop = () => {};
Object.defineProperty(window, `scrollTo`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `play`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `pause`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `load`, {value: noop, writable: true});
