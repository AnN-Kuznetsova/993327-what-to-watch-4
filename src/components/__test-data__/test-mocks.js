const promoMovie = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  releaseDate: new Date(2014, 0),
  posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundUrl: `img/bg-the-grand-budapest-hotel.jpg`,
};

const movieTitles = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `No Country for Old Men`,
  `Snatch`,
  `Moonrise Kingdom`,
  `Seven Years in Tibet`,
  `Midnight Special`,
  `War of the Worlds`,
  `Dardjeeling Limited`,
  `Orlando`,
  `Mindhunter`,
  `Midnight Special`,
];

const genreName = `Kids & Family`;

const movieTitle = `Aviator`;


export {
  genreName,
  movieTitle,
  movieTitles,
  promoMovie,
};
