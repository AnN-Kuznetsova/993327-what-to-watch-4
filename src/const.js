const NUMBER_OF_CARDS_IN_CATALOG_AT_STARTUP = 8;
const NUMBER_OF_CARDS_TO_INCREMENT = 8;
const MAX_FILTER_ELEMENTS_COUNT = 10;
const NUMBER_OF_SIMILAR_FILMS = 4;
const VISIBLE_PARTICIPANTS_COUNT = 4;
const DELAY_PLAYBACK_PREVIEW = 1000;
const MIN_REVIEW_TEXT_LENGTH = 50;
const MAX_REVIEW_TEXT_LENGTH = 400;
const RATING_RANGE = 5;
const ERROR_COLOR = `#a8421e`;

const BASENAME_FOR_DEPLOY = `/htmlacademy-what-to-watch`;

const AppRoute = {
  MAIN: BASENAME_FOR_DEPLOY + `/`,
  SIGN_IN: BASENAME_FOR_DEPLOY + `/login`,
  MY_LIST: BASENAME_FOR_DEPLOY + `/myList`,
  FILM: BASENAME_FOR_DEPLOY + `/films/:id`,
  ADD_REVIEW: BASENAME_FOR_DEPLOY + `/films/:id/review`,
  PLAYER: BASENAME_FOR_DEPLOY + `/player/:id`,
};


export {
  DELAY_PLAYBACK_PREVIEW,
  ERROR_COLOR,
  MAX_FILTER_ELEMENTS_COUNT,
  MAX_REVIEW_TEXT_LENGTH,
  MIN_REVIEW_TEXT_LENGTH,
  NUMBER_OF_CARDS_IN_CATALOG_AT_STARTUP,
  NUMBER_OF_CARDS_TO_INCREMENT,
  NUMBER_OF_SIMILAR_FILMS,
  RATING_RANGE,
  VISIBLE_PARTICIPANTS_COUNT,
  AppRoute,
  BASENAME_FOR_DEPLOY,
};
