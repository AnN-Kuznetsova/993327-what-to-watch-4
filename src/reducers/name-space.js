export const NameSpace = {
  DATA: `DATA`,
  APPLICATION: `APPLICATION`,
  USER: `USER`,
};
