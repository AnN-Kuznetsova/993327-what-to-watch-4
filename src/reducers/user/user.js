import {extend} from "../../utils/utils";


const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  loginError: null,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_LOGIN_ERROR: `SET_LOGIN_ERROR`,
};

const ActionCreator = {
  requireAuthorization: (status) => ({
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status,
  }),
  setLoginError: (error) => ({
    type: ActionType.SET_LOGIN_ERROR,
    payload: error,
  }),
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then(() => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.email,
      password: authData.password,
    })
    .then(() => {
      dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      dispatch(ActionCreator.setLoginError(null));
    })
    .catch((error) => {
      dispatch(ActionCreator.setLoginError(error));
    });
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload,
      });

    case ActionType.SET_LOGIN_ERROR:
      return extend(state, {
        loginError: action.payload
      });

    default:
      return state;
  }
};


export {
  ActionCreator,
  ActionType,
  AuthorizationStatus,
  Operation,
  reducer,
};
