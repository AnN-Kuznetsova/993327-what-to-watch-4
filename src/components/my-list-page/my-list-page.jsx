import PropTypes from "prop-types";
import React, {PureComponent} from "react";
import {connect} from "react-redux";

import {ActionCreator as ApplicationActionCreator} from "../../reducers/application/application";
import {Catalog} from "../catalog/catalog";
import {Footer} from "../footer/footer";
import {Header} from "../header/header";
import {PageType} from "../../const";


class MyListPageComponent extends PureComponent {
  componentDidMount() {
    this.props.onOpenMyListPage();
  }

  componentDidUpdate() {}

  render() {

    return (
      <div className="user-page">
        <Header />

        <section className="catalog">
          <h2 className="catalog__title visually-hidden">Catalog</h2>

          <Catalog />
        </section>

        <Footer />
      </div>
    );
  }
}


MyListPageComponent.propTypes = {
  onOpenMyListPage: PropTypes.func.isRequired,
};


const mapStateToProps = (state) => ({

});

const mapDispatchToProps = (dispatch) => ({
  onOpenMyListPage() {
    dispatch(ApplicationActionCreator.changeActivePage(PageType.MY_LIST));
  },
});

const MyListPage = connect(mapStateToProps, mapDispatchToProps)(MyListPageComponent);


export {
  MyListPageComponent,
  MyListPage,
};
