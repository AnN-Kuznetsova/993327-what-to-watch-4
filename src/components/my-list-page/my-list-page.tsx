
import * as React, {PureComponent} from "react";
import {connect} from "react-redux";

import {ActionCreator as ApplicationActionCreator} from "../../reducers/application/application";
import {ActionCreator as DataActionCreator, Operation} from "../../reducers/data/data";
import {Catalog} from "../catalog/catalog";
import {Footer} from "../footer/footer";
import {Header} from "../header/header";
import {PageType} from "../../const";
import {getDataError} from "../../reducers/data/selectors";


class MyListPageComponent extends PureComponent {
  componentDidMount() {
    this.props.onOpenMyListPage();
  }

  render() {
    const {dataError} = this.props;

    return (
      <div className="user-page">
        <Header />

        <section className="catalog">
          {dataError && <h2 className="catalog__title">Data loading error. Sorry</h2>}

          {!dataError &&
          <React.Fragment>
            <h2 className="catalog__title visually-hidden">Catalog</h2>
            <Catalog />
          </React.Fragment>}
        </section>

        <Footer />
      </div>
    );
  }
}


MyListPageComponent.propTypes = {
  onOpenMyListPage: PropTypes.func.isRequired,
  dataError: PropTypes.object,
};

const mapStateToProps = (state) => ({
  dataError: getDataError(state),
});

const mapDispatchToProps = (dispatch) => ({
  onOpenMyListPage() {
    dispatch(DataActionCreator.setDataError(null));
    dispatch(ApplicationActionCreator.changeActivePage(PageType.MY_LIST));
    dispatch(Operation.loadFavoriteMovies());
    dispatch(DataActionCreator.setMaxMoviesCount(null));
    dispatch(ApplicationActionCreator.resetVisibleMoviesCount());
  },
});

const MyListPage = connect(mapStateToProps, mapDispatchToProps)(MyListPageComponent);


export {
  MyListPageComponent,
  MyListPage,
};
