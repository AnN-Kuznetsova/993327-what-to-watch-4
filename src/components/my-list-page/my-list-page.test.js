import * as React from "react";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";

import {AuthorizationStatus} from "../../reducers/user/user";
import {MyListPageComponent} from "./my-list-page";
import {NameSpace} from "../../reducers/name-space";
import {PageType} from "../../const";
import {history} from "../../history";
import {mockMovies} from "../../__test-data__/test-mocks";


const mockStore = configureStore([]);
const store = mockStore({
  [NameSpace.DATA]: {
    movies: mockMovies,
  },
  [NameSpace.APPLICATION]: {
    visibleMoviesCount: 8,
    activePage: PageType.MY_LIST,
  },
  [NameSpace.USER]: {
    authorizationStatus: AuthorizationStatus.AUTH,
  },
});


describe(`Render MyListPage`, () => {
  it(`MyListPage should match with snapshot`, () => {
    const props = {
      onOpenMyListPage: () => {},
      dataError: null,
    };

    const myListPageSnapshot = renderer.create(
        <Router history={history} >
          <Provider store={store} >
            <MyListPageComponent {...props} />
          </Provider>
        </Router>
    ).toJSON();

    expect(myListPageSnapshot).toMatchSnapshot();
  });


  it(`MyListPage should match with snapshot with error`, () => {
    const props = {
      onOpenMyListPage: () => {},
      dataError: {status: 400},
    };

    const myListPageSnapshot = renderer.create(
        <Router history={history} >
          <Provider store={store} >
            <MyListPageComponent {...props} />
          </Provider>
        </Router>
    ).toJSON();

    expect(myListPageSnapshot).toMatchSnapshot();
  });
});
