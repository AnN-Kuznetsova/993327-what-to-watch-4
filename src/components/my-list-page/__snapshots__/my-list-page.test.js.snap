// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MyListPage MyListPage should match with snapshot 1`] = `
<div
  className="user-page"
>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header false user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link false"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "pointerEvents": "none",
          }
        }
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        href="/myList"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="/img/avatar.jpg"
            width="63"
          />
        </div>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "pointerEvents": "none",
          }
        }
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;

exports[`Render MyListPage MyListPage should match with snapshot with error 1`] = `
<div
  className="user-page"
>
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>
  <header
    className="page-header false user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link false"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "pointerEvents": "none",
          }
        }
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        href="/myList"
        onClick={[Function]}
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="/img/avatar.jpg"
            width="63"
          />
        </div>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title"
    >
      Data loading error. Sorry
    </h2>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
        style={
          Object {
            "pointerEvents": "none",
          }
        }
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        © 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
